webapp/src/components/modals/create_issue/index.tsx

```diff
@@ -0,0 +1,280 @@
+
+    // handle issue creation after form is populated
+    const handleCreate = useCallback(async (e: React.FormEvent<HTMLFormElement> | Event) => {
+        e.preventDefault();
+
+        if (!validator.validate() || !issueTitle) {            
+            setIssueTitleValid(Boolean(issueTitle));
+            setShowErrors(true);
+            return;
+        }
+
+        const postId = post?.id ?? '';
+
+        const issue = {
+            title: issueTitle,
+            description: issueDescription,
+            project_id: project?.project_id,
+            labels: labels.map((label) => label.value),
+            assignees: assignees.map((assignee) => assignee.value),
+            milestone: milestone?.value,
+            post_id: postId,
+            channel_id: channelId,
+        };
+
+        setSubmitting(true);
+
+        const created = await createIssue(issue)(dispatch);
+        if (created.error) {
+            const errMessage = getErrorMessage((created as {error: ErrorType}).error.message);
+            setError(errMessage);
+            setShowErrors(true);
+            setSubmitting(false);
+            return;
+        }
+
+        handleClose();
+    }, [issueTitle, issueTitleValid, issueDescription, validator, labels, milestone, assignees, project, channelId]);
+
+    const handleClose = useCallback(() => {
+        setError('');
+        setSubmitting(false);
+        setProject(null);
+        setIssueTitle('');
+        setIssueDescription('');
+        setLabels([]);
+        setAssignees([]);
+        setMilestone(null);
+        setShowErrors(false);
+        setIssueTitleValid(true);
+        dispatch(closeCreateIssueModal());
+    }, []);
+
+    const handleProjectChange = useCallback((project: ProjectSelection) => setProject(project), []);
+
+    const handleLabelsChange = useCallback((newLabels: OnChangeType) => setLabels(newLabels as SelectionType[]), []);
+
+    const handleAssigneesChange = useCallback((newAssignees: OnChangeType) => setAssignees(newAssignees as SelectionType[]), []);
+
+    const handleMilestoneChange = useCallback((newMilestone: OnChangeType) => setMilestone(newMilestone as SelectionType), []);
+
+    const handleIssueTitleChange = useCallback((issueTitle: string) => {        
+        setIssueTitle(issueTitle);
+        if (issueTitle && !issueTitleValid) {
+            setIssueTitleValid(true);
+        }
+    }, [issueTitleValid]);
+
+    const handleIssueDescriptionChange = useCallback((issueDescription: string) => setIssueDescription(issueDescription), []);
+
+    const issueAttributeSelectors = useMemo(() => {              
+        if (!project || !project?.project_id) {
+            return null;
+        }
+
+        const dropdownProps = {
+            projectID: project.project_id,
+            projectName: project.name,
+            theme: theme,
+        }
+
+        const labelProps = {
+            selectedLabels: labels,
+            onChange: handleLabelsChange,
```

Comment:
This is a case where we are not specifying a dependency in the dependency array for this `useMemo` callback. We should supply `handleLabelsChange` here. Even though `handleLabels` change doesn't change, having missing dependencies in the array makes it difficult to refactor in the future, to make sure we don't break anything.

Even if the optimizations were worth it, I think this would be less error-prone as far as dependency arrays:

```tsx
const dropdownProps = useMemo(() => ({
    projectID: project.project_id,
    projectName: project.name,
    theme: theme,
}), [project, theme]);

const labelSelector = useMemo(() => (
    <GitlabLabelSelector
        {...dropdownProps}
        selectedLabels={labels}
        onChange={handleLabelsChange}
    />
), [dropdownProps, labels, handleLabelsChange]);

const issueAttributeSelectors = (
    <>
        {labelSelector}
        // ...
    </>
);
```

But I think that having all of these optimizations makes the code harder to read and is error-prone. And (I know I keep repeating myself), I think for this particular feature the optimizations are not worth the risk that it brings.